kse/src/main/java/org/kse/crypto/publickey/JwkPublicKeyExporter.java

```diff
@@ -0,0 +1,187 @@
+import com.nimbusds.jose.util.Base64URL;
+import org.bouncycastle.jcajce.provider.asymmetric.edec.BCEdDSAPublicKey;
+import org.bouncycastle.jce.spec.ECNamedCurveSpec;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.jwk.Curve;
+import com.nimbusds.jose.jwk.ECKey;
+import com.nimbusds.jose.jwk.RSAKey;
+
+public class JwkPublicKeyExporter {
+    private final String alias;
+    private final KeyExporter keyExporter;
+
+    private JwkPublicKeyExporter(PublicKey publicKey, String alias) {
+        this.alias = alias;
+        if (publicKey instanceof ECPublicKey) {
+            this.keyExporter = new ECPublicKeyExporter(publicKey);
+        } else if (publicKey instanceof BCEdDSAPublicKey) {
+            this.keyExporter = new EdDSAPublicKeyExporter(publicKey);
+        } else if (publicKey instanceof RSAPublicKey) {
+            this.keyExporter = new RSAPublicKeyExporter(publicKey);
+        } else {
+            this.keyExporter = new NotSupportedKeyTypeExporter();
+        }
+    }
+
+    public static JwkPublicKeyExporter from(PublicKey publicKey, String alias) {
+        return new JwkPublicKeyExporter(publicKey, alias);
+    }
+
+    public boolean canExport() {
+        return this.keyExporter.supportsKey();
+    }
+
+    public byte[] get() throws JOSEException {
+        return keyExporter.exportWithAlias(alias);
+    }
+
+    interface KeyExporter {
+        byte[] exportWithAlias(String alias) throws JOSEException;
+
+        boolean supportsKey();
+    }
+
+    private static class NotSupportedKeyTypeExporter implements KeyExporter {
+        @Override
+        public byte[] exportWithAlias(String alias) {
+            return new byte[0];
+        }
+
+        @Override
+        public boolean supportsKey() {
+            return false;
+        }
+    }
+
+    private static class EdDSAPublicKeyExporter implements KeyExporter {
+        private final BCEdDSAPublicKey bcEdDSAPublicKey;
+        private static final Map<String, Curve> supportedCurvesMap =
+                Map.of(
+                        "Ed25519", Curve.Ed25519,
+                        "Ed448", Curve.Ed448
+                );
+
+        EdDSAPublicKeyExporter(PublicKey publicKey) {
+            this.bcEdDSAPublicKey = (BCEdDSAPublicKey) publicKey;
+        }
+
+        private Optional<Curve> getCurve(BCEdDSAPublicKey bcEdDSAPublicKey) {
+            return Optional.ofNullable(supportedCurvesMap.get(bcEdDSAPublicKey.getAlgorithm()));
+        }
+
+        @Override
+        public byte[] exportWithAlias(String alias) throws JOSEException {
+            Optional<Curve> maybeCurve = getCurve(bcEdDSAPublicKey);
+            if (maybeCurve.isEmpty()) {
+                return new byte[0];
+            }
+            Curve curve = maybeCurve.get();
+            OctetKeyPair.Builder builder = new OctetKeyPair.Builder(
+                    curve,
+                    Base64URL.encode(bcEdDSAPublicKey.getEncoded())
```

Comment:
That creates a wrong value for "x" in the JWK:
{
    "kty": "OKP",
    "crv": "Ed25519",
    "kid": "ed25519",
    "x": "MCowBQYDK2VwAyEAE-6D_zFv2Rcuc02erwnRDGUIjOtUDHLe4lTHg2Tlhzc"
}

The problem here is that the SubjectPublicKeyInfo ASN.1 structure around the key is still there. 

The field "x" should only contain the 32 bytes of the x coordinate of the key. You can extract those bytes with the following code:
```
SubjectPublicKeyInfo subjectPublicKeyInfo = SubjectPublicKeyInfo.getInstance(bcEdDSAPublicKey.getEncoded());
byte[] rawKey = subjectPublicKeyInfo.getPublicKeyData().getBytes();
```